service/task.go

```diff
@@ -332,15 +333,16 @@ func (uis *UIServer) taskPage(w http.ResponseWriter, r *http.Request) {
 				continue
 			}
 			execTasks = append(execTasks, *et)
+			execTaskIDs = append(execTaskIDs, t)
 		}
 		execTasks, err = task.MergeTestResultsBulk(execTasks, nil)
 		if err != nil {
 			uis.LoggedError(w, r, http.StatusInternalServerError, err)
 			return
 		}
-		for _, execTask := range execTasks {
+		for i, execTask := range execTasks {
 			uiTask.ExecutionTasks = append(uiTask.ExecutionTasks, uiExecTask{
-				Id:        execTask.Id,
+				Id:        execTaskIDs[i],
 				Name:      execTask.DisplayName,
 				TimeTaken: execTask.TimeTaken,
 				Status:    execTask.ResultStatus(),
```

Comment:
do we need to change the task id in the test results as well? (I can't tell what we use it for)

Reply:
Only the status and name are used, but I changed it for correctness's sake